TJCTF/Gamer_R/2020-24-05-Gamer_R.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: writeup
+title:  "Gamer R"
+date:   2020-24-05 20:00:00 +0200
+event: "TJCTF 2020"
+categories: [writeup]
+image: img/duck.png
+ctf_categories: [Reversing]
+author: "Trigleos"
+---
+
+# Gamer R
+
+
+## TL;DR
+Reverse a Unity game and patch it to get the flag
+## Description
+This challenge was the last Reversing challenge for TJCTF and gave 80 points. The challenge provided you with a folder that contained a binary as well as several Unity related files and libraries. The game itself was pretty simple. All you had to do was hit space to shoot a knife and hit an orange, all while trying to avoid hitting the ducks swimming randomly in between.
+If you manage to hit the orange, your score gets increased by 1.
+
+![game](img/game.gif)
+
+When hitting a duck, your score gets reset to zero,
+
+![gameover](img/gameover.gif)
+
+which means that we probably need to reach a certain score to get the flag. Unity games are normally written in C# which is an interpreted language that generates a bytecode instead of machine code which means that we can easily decompile it. I'll use dnSpy to decompile the program. 
+
+## Reversing
+Unity stores the game logic in *Projectname*_Data/Managed/Assembly-CSharp.dll so let's open it in dnSpy
+
+![dnSpy](img/dnSpy.png)
+
+Let's try to make the game easier by changing the duck collision behaviour described in the class DuckCollisionSystem.
+The important part of the code looks like this:
+````C#
```

Comment:
Use "csharp" for syntax highlighting instead of C#
https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers